napari/_qt/utils.py

```diff
@@ -323,3 +323,40 @@ def delete_qapp(app):
     # calling a second time is necessary on PySide2...
     # see: https://bugreports.qt.io/browse/PYSIDE-1470
     QApplication.instance()
+
+
+def get_view_direction_in_scene_coordinates(
```

Comment:
maybe this can go in a vispy utils file as it's more of a vispy thing than a qt thing. one day we might refactor the canvas into more of its own object and it'll make sense to go with that

Reply:
That sounds good! This was placed here temporarily just to get the ball rolling since we weren't really sure where it should go.

An alternative I had been considering is that instead of the vispy `ViewBox`, it takes just the screen center in canvas coordinates and the canvas-to-scene transform. In this case, transform is defined as an object with a `.map` (world to canvas) and `.imap` (canvas to world) method. In this way, it will work irrespective what we are using for the view/

Either way, the `qt` utils is not the right place for this 😆 !

Reply:
I've moved `get_view_direction_in_scene_coordinates()` to `_vispy/utils.py`